src/components/add-favorites/add-favorites.jsx

```diff
@@ -2,22 +2,8 @@ import { useState } from "react";
 import ToggleButton from 'react-bootstrap/ToggleButton';
 
 export const AddFavs = ({ movieID, logFav, userFavIDs }) => {
-    var buttonText;
-    console.log("Favs - " + userFavIDs);
-    // console.log("before " + JSON.parse(userFavIDs));
-    // const [userFavs, setUserFavs] = useState(JSON.parse(userFavIDs));
-
-    // var checkedValue = false;
-    // if (userFavs.find(favMovie => favMovie == movieID)) {
-    //     checkedValue = true;
-    // }
-    // const [checked, setChecked] = useState(checkedValue);
-
-    // if (checkedValue) {
-    //     buttonText = "Remove from Favorites";
-    // } else {
-    //     buttonText = "Add to Favorites";
-    // }
+    const isFavorite = userFavIDs.includes(movieID)
+    const buttonText =  isFavorite ? "Remove from Favorites" : "Add to Favorites";
```

Comment:
tl;dr - in React it's best to try and reference state either from component state (via `useState` or class state) or props instead of attempting to derive it outside of React lifecycle, which is completely unpredictable and goes agains the top-down data flow philosophy

Reply:
Always pull shared state high enough into component tree to share to any and all child components dependent on that state, then share the state via props if only going down one level or only within a bounded component context, otherwise leverage custom hooks or React context to share state amongst multiple disparate components declaratively and cleanly.